### PR TITLE
Alias Kedro IPython extension to kedro.ipython

### LINKED ISSUE #1763: Alias IPython extension
I'm fed up of (mis)typing `%load_ext kedro.extras.extensions.ipython`. Let's make everyone's lives a bit easier and alias it.

Previously we had said that `%load_ext kedro.ipython` would be an improvement, but I'm going to make a drastic suggestion and go even further to suggest `%load_ext kedro`. If you're doing `%load_ext` anyway then you know it's an IPython extension so the `ipython` part is just unnecessary extra characters.

### How?
Really just one line of code in `kedro/__init__.py` that says:
```
from .extras.extensions.ipython import load_ipython_extension
```
This should turn the module `kedro` into an IPython extension without affecting any of the existing kedro functionality.

### Is that it?
No! This is the important part. As per discussion we had as part of https://github.com/kedro-org/kedro/issues/712 we need to make sure that this `import` doesn't have undesired side effects, since the above line of code will run _every time anyone imports anything from `kedro`_. This means that any imports in the `extras -> extensions -> ipython` import chain will also be executed.

There are three possible risks here:
* the `import` tries to import something that is not a core dependency of kedro and the user doesn't have (_which includes IPython!_)
* the `import` imports loads of things that exist but makes things very slow
* the `import` has side effects that we don't want to be executed (e.g. importing `from kedro.framework.project import LOGGING` actually configures logging)

All three risks here should be mitigated against because in https://github.com/kedro-org/kedro/pull/1761 I already hid all the non-standard library imports inside functions as a guard. So, unless I missed something, what you need to do is write some tests to ensure that this does not get accidentally broken in the future. Something like these (will need some cunning mocking):
* `import kedro` does not try to import `IPython`
* `import kedro` does not do imports like `from kedro.framework.project import LOGGING`

Obviously manually test all this too, but the important thing here is writing the automated tests so it doesn't break in future if someone moves an import to be unguarded at the top level of the ipython.py file.

### Anything else to do?
Yes! Update all mentions in the codebase of `kedro.extras.extensions.ipython` to `kedro`. This is probably just docs and the `kedro ipython/jupyter` commands and should be easy. The hard thing here is the writing the tests.

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
## Description
Closes #1763.

With many apologies to @noklam for the original work done in #1794 trying to alias `kedro.extras.extensions.ipython` to just `kedro`, it turns out to be much better to alias as `kedro.ipython` as we originally thought. The pattern `%load_ext name.ipython` seems to be about as common as `%load_ext name`. Its only disadvantage is a few extra characters. The main advantage is that it feels cleaner, there's no breaking import linter contracts, and there's no danger that importing kedro could possibly import IPython or have other undesired side-effects like setting up logging.

> **Note**. We should delete `kedro/extras/extensions/` from `develop` after this has merged.

## Note to reviewers
I recommend reviewing this commit by commit since I moved a lot of code from one file to another. This will make it clear when it's just code that's moved or where I've actually changed something. The only real change to code itself is in https://github.com/kedro-org/kedro/pull/1837/commits/637053e2b7af505f17e0c5154875c4a20e450d94, where the definition of the line magic and imports and moved to top level.

## Development notes

Tested:
- [x] the new alias works
- [x] backwards compatibility of `kedro.extras.extensions.ipython`
- [x] packaging kedro still includes the logo png files thanks to MANIFEST.in
- [x] deprecation warning shows: image

Note that doing the imports this way round (`kedro.extras.extensions.ipython` imports `kedro.ipython` and not vice versa) is much neater because it will allow us to remove the file from `develop` straight away. This way, as we modify the IPython extension going forward, the file we'll be modifying is the new correct one rather than the old deprecated one.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1837"></a>

### CODE REVIEW

**Review by noklam — APPROVED** [review]
Thanks! I think `%load_ext kedro.ipython` is a better solution.

After hustling with the original proposal `%load_ext kedro`, I think we should be very careful about any top-level import alias i.e. `import kedro`. I am now more cautious about the idea in #712, due to the way that Python Import system works, it's very dangerous to put things in the top-level. It now make more sense to me why Django do `django.shortcuts` instead of just `django`.

**Review by MerelTheisenQB — APPROVED** [review]
Great work and thanks for explaining so clearly why just `kedro` as alias isn't such a good solution after all @noklam @AntonyMilneQB 👍 

Should we keep one test to check `%load_ext kedro.extras.extensions.ipython` still works until it's deprecated?

**Comment by AntonyMilneQB** [comment]
@MerelTheisenQB good thinking, done in [645ce1f](https://github.com/kedro-org/kedro/pull/1837/commits/645ce1faa73587c207c076c98f2dbf5531e5b33c)!

Actually I've just found out that most of the tests in this file haven't really doing anything useful for a long time. The mocking is all wrong and so not testing much... I'm going to make a new ticket to fix this up properly.

**Comment by noklam** [comment]
I found `test_logging.py` and `test_pipeline_registry.py` interfere with `test_ipython.py`, I am slightly worried that this is due to the manual deletion of `sys.modules` and it has some unexpected effect.

`pytest tests/ipython/test_ipython.py tests/framework/project/test_logging.py ` will pass
`pytest tests/framework/project/test_logging.py  tests/ipython/test_ipython.py ` will fail

File: tests/ipython/test_ipython.py [hunk 1]
```diff
@@ -62,21 +63,13 @@ def my_register_pipeline():
             return_value=my_register_pipeline,
         )
         mocker.patch("kedro.framework.startup.configure_project")
```
Comment (noklam) [inline tests/ipython/test_ipython.py:65]:
```suggestion
        mocker.patch("kedro.ipython.configure_project")
```

This is the wrong patch that causing most of the confusion.

Comment (AntonyMilneQB) [reply tests/ipython/test_ipython.py:65]: Ahh, if I change this then can I remove the `no pragma` and the tests pass?

Comment (noklam) [reply tests/ipython/test_ipython.py:65]: Let's address this altogether at #1839

**Pull request merged**